app/celery/tasks.py

```diff
@@ -740,11 +582,28 @@ def handle_batch_error_and_forward(
             # in the form of a dict
             if isinstance(template, tuple):
                 template = template[0]
-
-            save_fn.apply_async(
-                (service.id, notification_id, signed, None),
-                queue=choose_database_queue(template, service),
+            retry_msg = "{task} notification for job {job} row number {row} and notification id {notif} and max_retries are {max_retry}".format(
+                task=task.__name__,
+                job=notification.get("job", None),
+                row=notification.get("row_number", None),
+                notif=notification_id,
+                max_retry=task.max_retries,
             )
+            current_app.logger.warning("Retry " + retry_msg)
+            try:
+                # If >1 notification has failed, we want to make individual
+                # tasks to retry those notifications.
+                if len(signed_and_verified) != 1:
```

Comment:
Is that just me, I might be paranoid, I'd feel safer with

```suggestion
                if len(signed_and_verified) > 1:
```

😅

Reply:
I don't see why this would matter, but I can change it.